endpoints/api_config_manager.py

```diff
@@ -182,15 +182,18 @@ def lookup_rest_method(self, path, http_method):
         <method> is the descriptor as specified in the API configuration. -and-
         <params> is a dict of path parameters matched in the rest request.
     """
+    method_key = http_method.lower()
     with self._config_lock:
       for compiled_path_pattern, unused_path, methods in self._rest_methods:
-        match = compiled_path_pattern.match(path)
+        if method_key not in methods:
+          continue
+        candidate_method_info = methods[method_key]
```

Comment:
nit: could we maybe save one lookup by doing the following instead?

    candidate_method_info = methods.get(method_key)
    if candidate_method_info is None:
        continue